### ISSUE TITLE
enhance tests to check no processes are left behind

### ISSUE THREAD

**Issue by chavafg** [clearcontainers/tests#826]
We should check after each test that:
- there is no `qemu` or `CC` process left behind.
- `cc-runtime list` command returns what is expected. 

I think this is done for the docker integration tests, but for the rest we need to implement this.

**Comment by grahamwhaley** [comment]
Just for reference, you might be able to use or get some ideas from the code in one of the stability check scripts here:
https://github.com/clearcontainers/tests/blob/master/integration/stability/soak_parallel_rm.sh#L62

**Comment by jodh-intel** [comment]
Yep - it maybe those useful functions could be added to a `lib.sh`?

**Comment by grahamwhaley** [comment]
We can also check there are no disk or network 'mounts' left around as well.
For reference, that might have caught this earlier :-)
https://github.com/clearcontainers/runtime/issues/924

**Comment by grahamwhaley** [comment]
@chavafg @jodh-intel I was peeking into this. I'm thinking of adding Ginkgo `BeforeSuite` and `AfterSuite` funcs to https://github.com/clearcontainers/tests/blob/master/functional/main_test.go
and have them check similarly to https://github.com/clearcontainers/tests/blob/master/integration/stability/soak_parallel_rm.sh#L62, so that would be:

- correct number of containers running
- correct number of proxy/shim/qemu's/runtimes (none...) running
- correct number from `runtime list` command
- correct number of pods in the vc pod directory
- correct number of mounts (no docker mounts) after all tests have finished
- and possibly check the network namespaces after all tests have run as well

Q. do you think that makes sense, and is the right place to add the checks
RFC: I'm contemplating if I just code that all up, again, in golang, or if I just factor out the existing tests from the script above and then call the script from golang... hmmm...

**Comment by chavafg** [comment]
Hi @grahamwhaley, 

I think that it would be better to run the checks after every test (in the `AfterEach` function ) This way we could find which test/scenario is the one that is leaving processes behind. Everything else seems good to me. 

What do you say?

**Comment by grahamwhaley** [comment]
that's true. I was thinking of the higher level as it will be v.easy to forget to add them to all the tests in the future... soooo .... what I might do is add them both to the high level and the individual files, so if somebody does forget/miss a call in the future we should still capture the problem, just a little later.

**Comment by chavafg** [comment]
Agree.
